### PR TITLE
To Do Challenge submission - loliphant

### PR DESCRIPTION
**Pull request by lroliphant**

### CODE REVIEW

File: app/components/toDoListController.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+toDoTaskList.controller('ToDoListController', [function() {
+
```
Comment (ptolemybarnes) [inline app/components/toDoListController.js:2]: Please avoid random linebreaks

File: app/components/toDoListController.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  var self = this;
+  self.toDoList = [];
+
+  self.addToDo = function() {
+    if(!self.newToDo || self.newToDo === '') {
```
Comment (ptolemybarnes) [inline app/components/toDoListController.js:7]: Nice guard clause.

File: app/components/toDoListController.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      return;
+    }
+    self.toDoList.push( {'task': self.newToDo, 'completed': false} );
+    self.newToDo = '';
+  };
+
+  self.taskCounter = function () {
```
Comment (ptolemybarnes) [inline app/components/toDoListController.js:14]: I approve of this. :+1:

File: app/components/toDoListController.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    return self.toDoList.length;
+  };
+
+  self.deleteTask = function(item) {
```
Comment (ptolemybarnes) [inline app/components/toDoListController.js:18]: Funk star deluxe :star:

File: app/components/toDoListController.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    var index = self.toDoList.indexOf(item);
+    self.toDoList.splice(index, 1);
+  };
+
+// convert to a function using filter without underscore js library
+  self.removeCompleteTasks = function() {
+    self.toDoList = _.filter(self.toDoList, function(todo){
```
Comment (ptolemybarnes) [inline app/components/toDoListController.js:25]: Hmm did you pull in the whole underscore library to do one filter operation? Your better than that lollyphant! https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: index.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+var express = require('express');
```
Comment (ptolemybarnes) [inline index.js:1]: Nice one on using express

File: test/e2e/toDoListFeature.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      taskInput.sendKeys('Walk Basil');
+      addButton.click();
+    });
+
+    it('checkbox should be empty when task is created', function () {
+      expect(element(by.id('status')).getAttribute('class')).toEqual('completed-false');
+    });
+
+    it('should mark task as complete if checkbox is selected', function () {
+      completeTask.click();     expect(element(by.id('status')).getAttribute('class')).toEqual('completed-true');
```
Comment (ptolemybarnes) [inline test/e2e/toDoListFeature.js:46]: ooo don't have multiple lines on one line like this.

File: test/toDoListController.spec.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    expect(ctrl.toDo).toBeUndefined();
+  });
+
+  describe('is initialised with an empty list', function () {
+    it('so has no to dos yet', function() {
+      expect(ctrl.toDoList.length).toEqual(0);
+    });
+  });
+
+  var fakeTask = [
```
Comment (ptolemybarnes) [inline test/toDoListController.spec.js:20]: nice

**Comment by ptolemybarnes** [comment]
All things considered, pretty superb submission :checkered_flag:

File: app/components/toDoListController.js (hunk 5 above)
Comment (lroliphant) [inline app/components/toDoListController.js:25]: @ptolemybarnes, I did indeed, late on Sunday night, I felt pretty bad about it and if it helps it is in my readme to refactor/todo. Thanks for the link. I enjoyed this challenge so will be revisiting it and reworking.

**Pull request closed without merging**